Gems/Terrain/Code/CMakeLists.txt

```diff
@@ -16,6 +16,8 @@ ly_add_target(
         PRIVATE
             Source
     BUILD_DEPENDENCIES
+        PRIVATE
+            AZ::AtomCore
```

Comment:
TerrainSystem.h/cpp are the ones that use array_view. TerrainSystem.h is not a public header of the Terrain module so it only privately depends on AtomCore for its build.